interp/notation.ml

```diff
@@ -39,6 +39,30 @@ open Context.Named.Declaration
     expression, set this scope to be the current scope
 *)
 
+let notation_entry_eq s1 s2 = match (s1,s2) with
+| InConstrEntry, InConstrEntry -> true
+| InCustomEntry s1, InCustomEntry s2 -> String.equal s1 s2
+| (InConstrEntry | InCustomEntry _), _ -> false
+
+let notation_entry_level_eq s1 s2 = match (s1,s2) with
+| InConstrEntrySomeLevel, InConstrEntrySomeLevel -> true
+| InCustomEntryLevel (s1,n1), InCustomEntryLevel (s2,n2) -> String.equal s1 s2 && n1 = n2
+| (InConstrEntrySomeLevel | InCustomEntryLevel _), _ -> false
+
+let notation_eq (from1,ntn1) (from2,ntn2) =
+  notation_entry_level_eq from1 from2 && String.equal ntn1 ntn2
+
```

Comment:
Shouldn't this be in in constrexpr_ops ?

Reply:
This would indeed be consistent with type `notation` being in `constrexpr.ml`. But `Constrexpr_ops` depends on `Smartlocate` which depends on `Notation` what requires `Notation` before `Constrexpr_ops`. On the other side, `eq_notation` is used in `Notation` (e.g. for `scope_eq`). So, we would need to break the circularity. Maybe it is `Constrexpr_ops.is_constructor` (sole dependency of `Constrexpr_ops` into `Smartlocate` thus `Notation`), which is not at its right place, but then what to do with `Constrexpr_ops.cases_pattern_fold_names` which depends on `is_constructor` (making it abstract over the latter and pass it as an argument?).

In short, it is a bit of a headache...